react/src/pages/VariantQueryPage.tsx

```diff
@@ -145,10 +146,12 @@ const VariantQueryPage: React.FC<{}> = () => {
                         <Typography variant="subtitle" bold={!queryOptionsForm.ensemblId.value}>
                             Gene Name
                         </Typography>
-                        <Input
-                            disabled={!!queryOptionsForm.ensemblId.value}
-                            onChange={e => updateQueryOptionsForm('gene')(e.currentTarget.value)}
-                            value={queryOptionsForm.gene.value}
+                        <GeneSearch
+                            selectedGene={queryOptionsForm.gene.value}
+                            onSelect={geneOption => {
+                                updateQueryOptionsForm('gene')(geneOption.value.name);
+                                updateQueryOptionsForm('ensemblId')(geneOption.value.ensemblId);
```

Comment:
Say user has selected a gene (like APOE), the ensmbl id field will get autopopulated. When user edits the ensmbl id, I think we should clear out the gene name automatically as well because now the ensembl id and the gene name don't represent the same gene anymore.